tests/ansible/ocsp/tasks/certificate_self_validation_with_crl.yml

```diff
@@ -0,0 +1,575 @@
+    - dsctl slapd-localhost start
+
+- name: Configure secure ldap connection and enable client revocation check
+  community.docker.docker_container_exec:
+    container: "{{ ocsp_container }}"
+    command: "{{ item }}"
+  loop:
+    - pki-server stop
+    - pki-server ocsp-config-set internaldb.ldapconn.port 3636
+    - pki-server ocsp-config-set internaldb.ldapconn.secureConn true
+    - pki-server ocsp-config-set auths.revocationChecking.enabled true
+    - pki-server start --wait
+
+- name: Interact with good certificate
+  community.docker.docker_container_exec:
+    container: "{{ ocsp_container }}"
+    command: pki -n ocspadmin ocsp-user-show ocspadmin
+  register: ocsp_command
+  failed_when: "'User ID: ocspadmin' not in ocsp_command.stdout"
+
+- name: Identify the admin certificate serial
+  community.docker.docker_container_exec:
+    container: "{{ ca_container }}"
+    command: "pki -n caadmin  ca-cert-find --email ocspadmin@example.com"
+  register: ca_command
+
+- name: Put the OCSP admin on hold
+  community.docker.docker_container_exec:
+    container: "{{ ca_container }}"
+    command: "pki -n caadmin  ca-cert-hold {{ ca_command.stdout | regex_search('\\s*Serial Number:\\s*(\\S*)', '\\1') | first }}  --force"
+
+- name: Wait for CRL propagation
+  ansible.builtin.pause:
+    seconds: 15
+
+- name: Interact with revoked certificate
+  community.docker.docker_container_exec:
+    container: "{{ ocsp_container }}"
+    command: pki -n ocspadmin ocsp-user-show ocspadmin
+  register: ocsp_command
+  failed_when: "'PKIException: Unauthorized' not in ocsp_command.stderr"
+
+- name: Release the OCSP admin certificate
+  community.docker.docker_container_exec:
+    container: "{{ ca_container }}"
+    command: "pki -n caadmin  ca-cert-release-hold {{ ca_command.stdout | regex_search('\\s*Serial Number:\\s*(\\S*)', '\\1') | first }}  --force"
+
+- name: Wait for CRL propagation
+  ansible.builtin.pause:
+    seconds: 15
+
+- name: Interact with good certificate again
+  community.docker.docker_container_exec:
+    container: "{{ ocsp_container }}"
+    command: pki -n ocspadmin ocsp-user-show ocspadmin
+  register: ocsp_command
+  failed_when: "'User ID: ocspadmin' not in ocsp_command.stdout"
+
+- name: Identify the OCSP DS certificate serial
+  community.docker.docker_container_exec:
+    container: "{{ ca_container }}"
+    command: "pki -n caadmin  ca-cert-find --name {{ ocspds_hostname }}"
+  register: ca_command
+
+- name: Put the OCSP DS certificate on hold
+  community.docker.docker_container_exec:
+    container: "{{ ca_container }}"
+    command: "pki -n caadmin  ca-cert-hold {{ ca_command.stdout | regex_search('\\s*Serial Number:\\s*(\\S*)', '\\1') | first }}  --force"
+
+- name: Restart OCSP
+  community.docker.docker_container_exec:
+    container: "{{ ocsp_container }}"
+    command: pki-server restart --wait
+
+- name: Interact with good certificate again
```

Comment:
Should this say "Interact with revoked certificate" instead?

Reply:
The command is run with a good certificate (not revoked). However, the DS server has a revoked certificate. I have modified to make it more clear.